scripts/west_commands/build_helpers.py

```diff
@@ -111,18 +111,22 @@ def is_zephyr_build(path):
     build directory.
 
     "Valid" means the given path is a directory which contains a CMake
-    cache with a 'ZEPHYR_TOOLCHAIN_VARIANT' key.
+    cache with a 'ZEPHYR_BASE' or 'ZEPHYR_TOOLCHAIN_VARIANT' variable.
+
+    (The check for ZEPHYR_BASE introduced sometime after Zephyr 2.4 to fix
+    https://github.com/zephyrproject-rtos/zephyr/issues/28876; we
+    keep support for the second variable around for compatibility.)
```

Comment:
`ZEPHYR_BASE` was introduced here as a cached variable: https://github.com/zephyrproject-rtos/zephyr/pull/23054
So checking solely for `ZEPHYR_BASE` would allow `pristine` of Zephyr 2.3 and newer.
Keeping support for `ZEPHYR_TOOLCHAIN_VARIANT` means we keep support for `pristine` of 2.2 releases and older.